internal/framework/service/leaked_credential_check_rules/resource.go

```diff
@@ -0,0 +1,230 @@
+		return
+	}
+	// compare and create/delete accordingly
+	var createdRules []LCCRuleValueModel
+	toAdd, toRemove, toKeep := diffRules(plan.Rules, existing_rules)
+	// create
+	for _, createParam := range toAdd {
+		detection, err := r.client.V1.LeakedCredentialCheckCreateDetection(ctx, zoneID, createParam)
+		if err != nil {
+			resp.Diagnostics.AddError("Error creating a user-defined detection patter for Leaked Credential Check", err.Error())
+			return
+		}
+		createdRules = append(createdRules, buildLCCRuleValueModel(detection))
+	}
+	plan.Rules = createdRules // update plan rules with the newly created rules
+	// delete
+	for _, deleteParam := range toRemove {
+		_, err := r.client.V1.LeakedCredentialCheckDeleteDetection(ctx, zoneID, deleteParam)
+		if err != nil {
+			resp.Diagnostics.AddError("Error deleting a user-defined detection patter for Leaked Credential Check", err.Error())
+			return
+		}
+	}
+	// add the existing rules we kept to the plan, if any
+	for _, kr := range toKeep {
+		plan.Rules = append(plan.Rules, buildLCCRuleValueModel(kr))
+	}
+
+	diags = resp.State.Set(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+}
+
+// Delete all user-defined detection rules
+func (r *LeakedCredentialCheckRulesResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var state LeakedCredentialCheckRulesModel
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	zoneID := cloudflare.ZoneIdentifier(state.ZoneID.ValueString())
+	// fetch existing rules from API
+	rules, err := r.client.V1.LeakedCredentialCheckListDetections(ctx, zoneID, cloudflare.LeakedCredentialCheckListDetectionsParams{})
+	if err != nil {
+		resp.Diagnostics.AddError("Error fetching Leaked Credential Check user-defined detection patterns", err.Error())
+		return
+	}
+	for _, rule := range rules {
+		deleteParam := cloudflare.LeakedCredentialCheckDeleteDetectionParams{DetectionID: rule.ID}
+		_, err := r.client.V1.LeakedCredentialCheckDeleteDetection(ctx, zoneID, deleteParam)
+		if err != nil {
+			resp.Diagnostics.AddError("Error deleting a user-defined detection patter for Leaked Credential Check", err.Error())
+			return
+		}
+
+	}
+
+}
+
+func (r *LeakedCredentialCheckRulesResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	// req.ID is the zoneID for which you want to import the state of the
+	// Leaked Credential Check Rules
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("zone_id"), req.ID)...)
+	return
+}
+
+func diffRules(desired []LCCRuleValueModel, current []cloudflare.LeakedCredentialCheckDetectionEntry) (toAdd []cloudflare.LeakedCredentialCheckCreateDetectionParams, toRemove []cloudflare.LeakedCredentialCheckDeleteDetectionParams, toKeep []cloudflare.LeakedCredentialCheckDetectionEntry) {
```

Comment:
we'd like to avoid needing this is v5. is there a reason why we don't have the API modelled in a way that this is inbuild and uses a PUT/DELETE correctly?